scout/server/blueprints/cases/controllers.py

```diff
@@ -105,41 +134,87 @@ def case(store, institute_obj, case_obj):
+        'dismissed_detailed': 'dismiss_variant',
+        'commented_detailed': 'is_commented',
+    }
+    data = case_obj
+    
+    # Add the case comments
+    data['comments'] = store.events(institute_obj, case=case_obj, comments=True)
 
     data['manual_rank_options'] = MANUAL_RANK_OPTIONS
     data['dismissed_options'] = DISMISS_VARIANT_OPTIONS
     data['genetic_models'] = dict(GENETIC_MODELS)
     data['report_created_at'] = datetime.datetime.now().strftime("%Y-%m-%d %H:%M")
 
+    evaluated_variants = {}
+    for vt in variant_types:
+        evaluated_variants[vt] = []
     # We collect all causatives and suspected variants
+    # These are handeled in separate since they are on case level
     for var_type in ['causatives', 'suspects']:
-        # data[var_type] can include strings so we need to be careful
-        # remove any element which is not a dictionary
-        raw = [var for var in data[var_type] if isinstance(var, dict)]
-        # get detailed info for the causatives and add to data
-        var_type += '_detailed'
-        data[var_type] = variants_filter_by_field(store, raw, '_id', case_obj, institute_obj)
+        #These include references to variants
+        vt = '_'.join([var_type, 'detailed'])
+        for var_id in case_obj.get(var_type,[]):
+            variant_obj = store.variant(var_id)
+            if not variant_obj:
+                continue
+            # If the variant exists we add it to the evaluated variants
+            evaluated_variants[vt].append(variant_obj)
 
     ## get variants for this case that are either classified, commented, tagged or dismissed.
-    evaluated_variants = store.evaluated_variants(case_id=case_obj['_id'])
-
-    # get complete info for the acmg classified variants
-    data['classified_detailed'] = variants_filter_by_field(store, evaluated_variants,
-                                                           'acmg_classification', case_obj, institute_obj)
-
-    # get complete info for tagged variants
-    data['tagged_detailed'] = variants_filter_by_field(store, evaluated_variants,
-                                                       'manual_rank', case_obj, institute_obj)
-
-    # get complete info for dismissed variants
-    data['dismissed_detailed'] = variants_filter_by_field(store, evaluated_variants,
-                                                        'dismiss_variant', case_obj, institute_obj)
-
-    data['commented_detailed'] = variants_filter_by_field(store, evaluated_variants,
-                                                        'is_commented', case_obj, institute_obj)
+    for var_obj in store.evaluated_variants(case_id=case_obj['_id']):
+        # Check which category it belongs to
+        for vt in variant_types:
+            keyword = variant_types[vt]
+            # When found we add it to the categpry
+            if keyword in var_obj:
+                evaluated_variants[vt].append(var_obj)
+                break
```

Comment:
This break takes care of not having the same variant in two categories of the evaluated variants, right? I think we should instead add to the evaluated category all those that belong to them, regardless. That's because for the clinicians might be important to know that a variant that for instance is ACMG classified is already printed among the pinned, for instance. Otherwise they'll not find variants that have actions associated and might wonder why.